[BUG] NullPointerException during App export

**API-Mgr. CLI version**
1.0.0

**API-Manager and Service-Pack Version**
7.6.2 SP4

**Expected behavior**


**Actual behavior**
As OrgAdmin, as NullPointerException happens with the following call
`apim app get -s <myStage>-ultra`

**Log-Output in Debug**
```
2156   INFO     APIManagerAdapter| Successfully connected to API-Manager (7.6.2 SP4) on: ...
2303   WARN  rOrganizationAdapter| Using OrgAdmin only to load all organizations.
2364   ERROR ApplicationExportApp| Can't initialize API-Manager API-Representation.
com.axway.apim.lib.errorHandling.AppException: Can't initialize API-Manager API-Representation.
        at com.axway.apim.adapter.clientApps.APIMgrAppsAdapter.getApplications(APIMgrAppsAdapter.java:143)
        at com.axway.apim.appexport.ApplicationExportApp.runExport(ApplicationExportApp.java:76)
        at com.axway.apim.appexport.ApplicationExportApp.export(ApplicationExportApp.java:55)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.axway.apim.cli.APIManagerCLI.run(APIManagerCLI.java:138)
        at com.axway.apim.cli.APIManagerCLI.main(APIManagerCLI.java:76)
Caused by: java.lang.NullPointerException
        at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:889)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3005)
        at com.axway.apim.adapter.apis.APIManagerQuotaAdapter.getQuotaForAPI(APIManagerQuotaAdapter.java:103)
        at com.axway.apim.adapter.clientApps.APIMgrAppsAdapter.getApplications(APIMgrAppsAdapter.java:135)
        ... 8 more
```

--- a/modules/apim-adapter/src/main/java/com/axway/apim/adapter/apis/APIManagerAPIAccessAdapter.java
+++ b/modules/apim-adapter/src/main/java/com/axway/apim/adapter/apis/APIManagerAPIAccessAdapter.java
@@ -181,16 +181,23 @@ public class APIManagerAPIAccessAdapter {
 			RestAPICall request = new POSTRequest(entity, uri, APIManagerAdapter.hasAdminAccount());
 			request.setContentType("application/json");
 			httpResponse = request.execute();
-			int statusCode = httpResponse.getStatusLine().getStatusCode();
+			int statusCode = httpResponse.getStatusLine().getStatusCode();			
 			if(statusCode < 200 || statusCode > 299){
 				String response = EntityUtils.toString(httpResponse.getEntity());
-				if(statusCode==409 && response.contains("resource already exists")) {
-					LOG.warn("API Access: "+apiAccess+" for " + type + " with ID: " + parentId + " already exists. Ignoring this error.");
-					return apiAccess;
+				if(statusCode==403 && response.contains("Unknown API")) {
+					LOG.warn("Got unexpected error: 'Unknown API' while creating API-Access ... Try again in 1 second.");
+					Thread.sleep(1000);
+					httpResponse = request.execute();
+					statusCode = httpResponse.getStatusLine().getStatusCode();
+					if(statusCode < 200 || statusCode > 299){
+						LOG.error("Error creating/updating API Access: "+apiAccess+". Response-Code: "+statusCode+". Got response: '"+response+"'");
+						throw new AppException("Error creating/updating API Access. Response-Code: "+statusCode+"", ErrorCode.API_MANAGER_COMMUNICATION);
+					} else {
+						LOG.info("Successfully created API-Access on retry. Received Status-Code: " +statusCode );
+					}
 				}
-				LOG.error("Error creating/updating API Access: "+apiAccess+". Response-Code: "+statusCode+". Got response: '"+response+"'");
-				throw new AppException("Error creating/updating API Access. Response-Code: "+statusCode+"", ErrorCode.API_MANAGER_COMMUNICATION);
 			}
+			
 			String response = EntityUtils.toString(httpResponse.getEntity());
 			apiAccess =  mapper.readValue(response, APIAccess.class);
 			// Clean cache for this ID (App/Org) to force reload next time

--- a/modules/apim-adapter/src/main/java/com/axway/apim/adapter/apis/APIManagerQuotaAdapter.java
+++ b/modules/apim-adapter/src/main/java/com/axway/apim/adapter/apis/APIManagerQuotaAdapter.java
@@ -97,6 +97,7 @@ public class APIManagerQuotaAdapter {
 	}
 	
 	public APIQuota getQuotaForAPI(String quotaId, String apiId) throws AppException {
+		if(!APIManagerAdapter.hasAdminAccount()) return null;
 		readQuotaFromAPIManager(quotaId);
 		APIQuota quotaConfig;
 		try {
@@ -110,6 +111,7 @@ public class APIManagerQuotaAdapter {
 	}
 	
 	public APIQuota saveQuota(APIQuota quotaConfig, String quotaId) throws AppException {
+		if(!APIManagerAdapter.hasAdminAccount()) return null;
 		URI uri;
 		HttpEntity entity;
 		HttpResponse httpResponse = null;
@@ -137,6 +139,7 @@ public class APIManagerQuotaAdapter {
 	}
 	
 	public APIQuota getDefaultQuota(Quota quotaType) throws AppException {
+		if(!APIManagerAdapter.hasAdminAccount()) return null;
 		readQuotaFromAPIManager(quotaType.getQuotaId());
 		APIQuota quotaConfig;
 		try {

--- a/modules/app-export/src/test/java/com/axway/apim/appexport/impl/ConsoleAppExportTest.java
+++ b/modules/app-export/src/test/java/com/axway/apim/appexport/impl/ConsoleAppExportTest.java
@@ -1,0 +1,68 @@
+package com.axway.apim.appexport.impl;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.cli.ParseException;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.axway.apim.adapter.apis.APIManagerMockBase;
+import com.axway.apim.api.model.apps.ClientApplication;
+import com.axway.apim.appexport.lib.AppExportCLIOptions;
+import com.axway.apim.appexport.lib.AppExportParams;
+import com.axway.apim.lib.errorHandling.AppException;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class ConsoleAppExportTest extends APIManagerMockBase {
+
+	private static final String TEST_PACKAGE = "com/axway/apim/appexport/apps/resultSet/";
+
+	ObjectMapper mapper = new ObjectMapper();
+
+	@BeforeClass
+	public void setTest() throws AppException, IOException {
+		setupMockData();
+		mapper.disable(MapperFeature.USE_ANNOTATIONS);
+		mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+	}
+
+	@Test
+	public void runStandardConsoleAppExport()
+			throws JsonParseException, JsonMappingException, IOException, AppException, ParseException {
+		List<ClientApplication> apps = mapper.readValue(this.getClass().getClassLoader().getResourceAsStream(
+				TEST_PACKAGE + "appsAsOrgAdmin.json"), new TypeReference<List<ClientApplication>>() {
+				});
+
+		AppExportParams cmdParams = new AppExportParams(new AppExportCLIOptions(new String[] {}));
+		ConsoleAppExporter consoleExp = new ConsoleAppExporter(cmdParams);
+		consoleExp.export(apps);
+	}
+
+	@Test
+	public void runWideConsoleAppExport()
+			throws JsonParseException, JsonMappingException, IOException, AppException, ParseException {
+		List<ClientApplication> apps = mapper.readValue(this.getClass().getClassLoader().getResourceAsStream(
+				TEST_PACKAGE + "appsAsOrgAdmin.json"), new TypeReference<List<ClientApplication>>() {
+				});
+
+		AppExportParams cmdParams = new AppExportParams(new AppExportCLIOptions(new String[] { "-wide" }));
+		ConsoleAppExporter consoleExp = new ConsoleAppExporter(cmdParams);
+		consoleExp.export(apps);
+	}
+
+	@Test
+	public void runUltraConsoleAppExport()
+			throws JsonParseException, JsonMappingException, IOException, AppException, ParseException {
+		List<ClientApplication> apps = mapper.readValue(this.getClass().getClassLoader().getResourceAsStream(TEST_PACKAGE + "appsAsOrgAdmin.json"), new TypeReference<List<ClientApplication>>() {});
+
+		AppExportParams cmdParams = new AppExportParams(new AppExportCLIOptions(new String[] { "-ultra" }));
+		ConsoleAppExporter consoleExp = new ConsoleAppExporter(cmdParams);
+		consoleExp.export(apps);
+	}
+}

--- a/modules/app-export/src/test/resources/com/axway/apim/appexport/apps/resultSet/appsAsOrgAdmin.json
+++ b/modules/app-export/src/test/resources/com/axway/apim/appexport/apps/resultSet/appsAsOrgAdmin.json
@@ -1,0 +1,78 @@
+[
+  {
+    "id": "68d1b0dc-0ec6-43c0-ba70-1ac8d5c3e7fe",
+    "name": "Wearable - Sleep Tracker",
+    "description": "Smart device app monitoring patient sleep cycle",
+    "email": "plexus.sleep.contact@plexorg.com",
+    "phone": null,
+    "enabled": true,
+    "state": "approved",
+    "oauthClientId": null,
+    "extClientId": null,
+    "apiKey": null,
+    "credentials": [
+      
+    ],
+    "appQuota": null,
+    "organization": {
+      "id": "5d4dfe67-783d-4d4b-8a9a-3a08635cf579",
+      "name": "Community",
+      "description": "Unverified untrusted developer users. Not intended for production-level client applications",
+      "email": null,
+      "image": null,
+      "restricted": "true",
+      "virtualHost": null,
+      "phone": null,
+      "enabled": "true",
+      "development": false,
+      "dn": "o=Community,ou=organizations,ou=APIPortal",
+      "createdOn": "1556038716879",
+      "startTrialDate": null,
+      "endTrialDate": null,
+      "trialDuration": null,
+      "isTrial": null
+    },
+    "image": "/api/portal/v1.3/applications/68d1b0dc-0ec6-43c0-ba70-1ac8d5c3e7fe/image",
+    "apis": [
+      
+    ]
+  },
+  {
+    "id": "83d59ccd-c507-4706-b586-199ff7ab69f7",
+    "name": "Wearable - Activity Tracker",
+    "description": "Smart device app monitoring patient heart rate and activity tracking",
+    "email": "plexus.activity.contact@careconnex.com",
+    "phone": null,
+    "enabled": true,
+    "state": "approved",
+    "oauthClientId": null,
+    "extClientId": null,
+    "apiKey": null,
+    "credentials": [
+      
+    ],
+    "appQuota": null,
+    "organization": {
+      "id": "5d4dfe67-783d-4d4b-8a9a-3a08635cf579",
+      "name": "Community",
+      "description": "Unverified untrusted developer users. Not intended for production-level client applications",
+      "email": null,
+      "image": null,
+      "restricted": "true",
+      "virtualHost": null,
+      "phone": null,
+      "enabled": "true",
+      "development": false,
+      "dn": "o=Community,ou=organizations,ou=APIPortal",
+      "createdOn": "1556038716879",
+      "startTrialDate": null,
+      "endTrialDate": null,
+      "trialDuration": null,
+      "isTrial": null
+    },
+    "image": "/api/portal/v1.3/applications/83d59ccd-c507-4706-b586-199ff7ab69f7/image",
+    "apis": [
+      
+    ]
+  }
+]

